apps/mobile/src/features/balances/stacks/sip10-balance.tsx

```diff
@@ -1,37 +1,51 @@
-import { TokenIcon } from '@/components/widgets/tokens/token-icon';
 import {
   useSip10AccountBalance,
   useSip10TotalBalance,
 } from '@/queries/balance/sip10-balance.query';
 
 import { Money } from '@leather.io/models';
-import { PressableProps } from '@leather.io/ui/native';
+import { Avatar } from '@leather.io/ui/native';
 
 import { TokenBalance } from '../token-balance';
 
-const sip10MaxDisplay = 3;
-interface Sip10TokenBalanceProps extends PressableProps {
+function getFallbackAvatar(contractId: string) {
+  // TODO LEA-2264 use avatars from Alex API
+  // extension uses StacksAssetAvatar and DynamicColorCircle for this
+  return `https://avatar.vercel.sh/${contractId}?size=36`;
+}
+
+interface Sip10TokenBalanceProps {
   availableBalance: Money;
+  contractId: string;
   fiatBalance: Money;
-  symbol: string;
+  imageCanonicalUri: string;
   name: string;
+  symbol: string;
 }
 export function Sip10TokenBalance({
   availableBalance,
+  contractId,
   fiatBalance,
+  imageCanonicalUri,
   name,
   symbol,
-  ...rest
 }: Sip10TokenBalanceProps) {
   return (
     <TokenBalance
       ticker={symbol}
-      icon={<TokenIcon ticker={symbol} />} // TODO LEA-1909: add images from uri
+      icon={
+        <Avatar
+          image={imageCanonicalUri !== '' ? imageCanonicalUri : getFallbackAvatar(contractId)}
```

Comment:
Doing this as sometimes the API returns `''` for the URI

On the `extension` we show them like this with a gradient fallback and a letter
![Screenshot 2025-03-07 at 16 03 52](https://github.com/user-attachments/assets/03c8f50a-4fa4-4e7b-8e89-460d7460b837)

I'll soon update this to use real icons from ALEX which I add to the `sip10` balance service